hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/ReplaceAliasesWithIDsInOp.java

```diff
@@ -0,0 +1,101 @@
+
+import static com.hedera.node.app.service.token.impl.handlers.transfer.EnsureAliasesStep.isAlias;
+import static java.util.Collections.emptyList;
+
+import com.hedera.hapi.node.base.AccountAmount;
+import com.hedera.hapi.node.base.NftTransfer;
+import com.hedera.hapi.node.base.TokenTransferList;
+import com.hedera.hapi.node.base.TransferList;
+import com.hedera.hapi.node.token.CryptoTransferTransactionBody;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Replaces aliases with IDs in the crypto transfer operation.
+ */
+public class ReplaceAliasesWithIDsInOp {
+    public ReplaceAliasesWithIDsInOp() {}
+
+    public CryptoTransferTransactionBody replaceAliasesWithIds(
+            final CryptoTransferTransactionBody op, final TransferContextImpl transferContext) {
+        final var resolutions = transferContext.resolutions();
+        final var replacedAliasesOp = CryptoTransferTransactionBody.newBuilder();
+        final var transferList = TransferList.newBuilder();
+        final var tokenTransfersList = new ArrayList<TokenTransferList>();
+        final var accountAmounts = new ArrayList<AccountAmount>();
+        // replace all aliases in hbar transfers
+        for (final var aa : op.transfers().accountAmountsOrElse(emptyList())) {
+            if (isAlias(aa.accountIDOrThrow())) {
+                final var resolvedId = resolutions.get(aa.accountID().alias());
+                accountAmounts.add(aa.copyBuilder().accountID(resolvedId).build());
+            } else {
+                accountAmounts.add(aa);
+            }
+        }
+        transferList.accountAmounts(accountAmounts);
+        replacedAliasesOp.transfers(transferList);
+
+        // replace all aliases in token transfers
+        for (final var adjust : op.tokenTransfersOrElse(emptyList())) {
+            final var tokenTransferList = TokenTransferList.newBuilder().token(adjust.token());
+            final List<AccountAmount> replacedTokenAdjusts = new ArrayList<>();
+            for (final var tokenAdjust : adjust.transfersOrElse(emptyList())) {
+                if (isAlias(tokenAdjust.accountIDOrThrow())) {
+                    final var resolvedId =
+                            resolutions.get(tokenAdjust.accountID().alias());
+                    replacedTokenAdjusts.add(
+                            tokenAdjust.copyBuilder().accountID(resolvedId).build());
+                } else {
+                    replacedTokenAdjusts.add(tokenAdjust);
+                }
+            }
+            if (!replacedTokenAdjusts.isEmpty()) {
+                tokenTransferList.transfers(replacedTokenAdjusts);
+            }
+
+            final List<NftTransfer> replacedNftAdjusts = new ArrayList<>();
+            for (final var nftAdjust : adjust.nftTransfersOrElse(emptyList())) {
+                final var nftAdjustCopy = nftAdjust.copyBuilder();
+                if (isAlias(nftAdjust.receiverAccountIDOrThrow()) || isAlias(nftAdjust.senderAccountIDOrThrow())) {
```

Comment:
These booleans can be stored as vars so the same check doesn't have to be repeated in the 'ifs' below. Just a thought, might be easier to read. Or maybe not, it's a lot either way 😂

Reply:
yeah done 😂